astropy/table/tests/test_table.py

```diff
@@ -3533,3 +3533,15 @@ def test_table_columns_update_deprecation():
         ),
     ):
         Table().columns.update({"a": [0]})
+
+
+def test_qtable_with_explicit_units():
+    # Regression test for gh-17047; the problem was that the dimensionless
+    # unit ended up being compared to np.ma.masked.  See also
+    # astropy/units/tests/test_units.py::test_comparison_dimensionless_with_np_ma_masked
+    tt = QTable(
+        data=[[1.0, 2.0, 3.0]],
+        names=["weight"],
+        units={"weight": u.dimensionless_unscaled},
+    )
```

Comment:
This could fit on a single line:
```suggestion
    tt = QTable({"a": [1.0, 2.0, 3.0]}, units={"a": u.one})
```

Reply:
Thanks, better indeed. Two changes holding on is enough, so I pushed it up.